src/precice/impl/ParticipantImpl.cpp

```diff
@@ -1414,6 +1425,7 @@ void ParticipantImpl::mapInitialWrittenData()
 void ParticipantImpl::mapWrittenData(std::optional<double> after)
 {
   PRECICE_TRACE();
+  Event e("mapWriteData", profiling::Fundamental);
```

Comment:
What do we expect if there is no write-mapping defined for this participant?

Should there be an event of size 0 or no event at all?

Reply:
> Should there be an event of size 0 or no event at all?

No event at all IMO.

> Another approach to these events would be to use mapping for all of them. This way the mapping event itself is more like a logical group of events.

Sounds good to me. Also `"mapWriteData"` would be a collection if there are two write data mappings, say one NN and one RBF.

Reply:
Changed the PR to avoid mapping events for non-existent read or write mappings.

Also used `mapping` for all of them.